feat: support stored procedure execution

gmendozamb opened the pull request:
Added the possibility to execute stored procedures by adding the exec word to the conditions and using raw_sql() from ibis instead of sql(). 

Also added the possibility to use Current Date as variable (@Today) in queries. It will be replaced by the current server date in the following format ('YYYY-MM-DD').

Use:
`EXEC usp_report_Sales @valuationDate = @Today, @ShowSerialNumbers=1`

nextchamp-saqib commented:
> Also added the possibility to use Current Date as variable (@today) in queries. It will be replaced by the current server date in the following format ('YYYY-MM-DD').

This might not be necessary. Can you check if `GETDATE()` function can be used to get the current date?

gmendozamb commented:
I tried and it didn’t work.

Le mar. 18 févr. 2025 à 03:57, Saqib Ansari ***@***.***> a
écrit :

> Also added the possibility to use Current Date as variable ***@***.***
> <https://github.com/today>) in queries. It will be replaced by the
> current server date in the following format ('YYYY-MM-DD').
>
> This might not be necessary. Can you check if GETDATE() function can be
> used to get the current date?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/frappe/insights/pull/437#issuecomment-2664987445>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AF75YSEPCW2ENYYUDUL2XQT2QLYYHAVCNFSM6AAAAABWPKEHPSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMNRUHE4DONBUGU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
> [image: nextchamp-saqib]*nextchamp-saqib* left a comment
> (frappe/insights#437)
> <https://github.com/frappe/insights/pull/437#issuecomment-2664987445>
>
> Also added the possibility to use Current Date as variable ***@***.***
> <https://github.com/today>) in queries. It will be replaced by the
> current server date in the following format ('YYYY-MM-DD').
>
> This might not be necessary. Can you check if GETDATE() function can be
> used to get the current date?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/frappe/insights/pull/437#issuecomment-2664987445>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AF75YSEPCW2ENYYUDUL2XQT2QLYYHAVCNFSM6AAAAABWPKEHPSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMNRUHE4DONBUGU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

gmendozamb commented:
> > Also added the possibility to use Current Date as variable (@today) in queries. It will be replaced by the current server date in the following format ('YYYY-MM-DD').
> 
> This might not be necessary. Can you check if `GETDATE()` function can be used to get the current date?

I tried with GETDATE() but it doesn't work. Everything is working as expected with your changes.
